prov/util/src/util_attr.c

```diff
@@ -398,17 +401,23 @@ int ofi_check_mr_mode(uint32_t api_version, uint32_t prov_mode,
 
 		switch (user_mode) {
 		case FI_MR_UNSPEC:
-			return !prov_mode || (prov_mode == OFI_MR_BASIC_MAP) ?
+			return OFI_CHECK_MR_SCALABLE(prov_mode) ||
+				OFI_CHECK_MR_BASIC(prov_mode) ?
 				0 : -FI_ENODATA;
 		case FI_MR_BASIC:
-			return prov_mode == OFI_MR_BASIC_MAP ? 0 : -FI_ENODATA;
+			return OFI_CHECK_MR_BASIC(prov_mode) ? 0 : -FI_ENODATA;
 		case FI_MR_SCALABLE:
-			return !prov_mode ? 0 : -FI_ENODATA;
+			return OFI_CHECK_MR_SCALABLE(prov_mode) ? 0 : -FI_ENODATA;
 		default:
 			return -FI_ENODATA;
 		}
 	} else {
-		return ((user_mode & prov_mode) == prov_mode) ? 0 : -FI_ENODATA;
+		if ((user_mode & FI_MR_BASIC) || !(prov_mode & FI_MR_BASIC))
+			return ((user_mode & prov_mode) == prov_mode) ?
+				0 : -FI_ENODATA;
```

Comment:
If prov_mode = BASIC | VIRT_ADDR | PROV_KEY | ALLOCATED, but user_mode = BASIC, this will fail.  I think we may want to assume that user_mode = BASIC is equivalent to setting the basic mr_mode bits.  Not sure if this is correct:
```
if (user_mode & FI_MR_BASIC) {  /* user wants BASIC */
    if (!OFI_CHECK_MR_BASIC(prov_mode)) /* provider does not support basic */
        return -FI_ENODATA;
    /* other potential mode bits must also be set */
    if ((user_mode & prov_mode & ~OFI_MR_BASIC_MAP) ==
       (prov_mode & ~OFI_MR_BASIC_MAP))
        return 0;
    return -FI_ENODATA;
}
```